### ISSUE TITLE
Add Instance Shared Parameter Value in Exported Filename

### ISSUE THREAD

**Issue by SteelcoreSystems** [jeremytammik/ExportCncFab#1]
I would like to know how to add an additional shared parameter instance value to the exported SAT/DXF filenames. The current SAT/DXF filename method only exports the "host level name_parentId_partId"

Only exporting the Revit element id makes it difficult and time consuming to determine what part was exported after exporting many elements. I have added an instance shared parameter to my Revit models and assigned it to parts.  The goal is to have this instance shared parameter value included within each exported filename.

When exporting, I get the following default file name:
     "Level Name_136667_136683.sat".

I would like for the file name to be exported with the "SortMark" instance value:
     "Level Name_"**SortMark**"_136667_136683.sat
     "Level 03_**SB-10**_13667_136683.sat"

For some reason, I am not able to output the desired filename with the "SortMark" instance value. Instead, I receive: 

"Level Name_**Autodesk.Revit.DB.Parameter**_136667_136686.sat"  

Here is an image of the modified code:
![2020-11-30_17h05_42](https://user-images.githubusercontent.com/32565599/100753215-1d5f2480-33af-11eb-8e36-9332ae92d39c.png)

Here is a link to the CncFabExporter 2020 version modified for floor parts.  

https://www.dropbox.com/sh/0b7io0feagzxwre/AABCnwrOSuScKrGvgJlEJvE2a?dl=0

Thank you so much for your time to read this message and providing any available input.

**Comment by jeremytammik** [comment]
I looked at your dropbox version and see the modifications you made to select floors instead of walls. They look fine to me.

The reason you get the result you see is because you are outputting the `Parameter` object, not the string it contains.

All you need to do is add a call to `AsString`.

I cleaned up your approach to avoid using the suboptimal `LookupParameter` method.

I implemented support for a new optional shared parameter named `CncFabSortMark` in the export parameters class instead.

If it is present on an element, it is included in the output filename.

I have not tested the code, so please let us know whether it works as expected. Thank you!

**Comment by jeremytammik** [comment]
Here is the diff for adding the sort mark:

https://github.com/jeremytammik/ExportCncFab/compare/2021.0.0.0...2021.0.0.1

**Comment by SteelcoreSystems** [comment]
Jeremy,  
Great Work!   I made the adjustments and on the first try everything exported perfectly.   
I greatly appreciate you making it easy to assign any parameter name for value exporting.   

Under the ExportCncFab/ExportParameters.cs file at line 32, I was able to change    
const string _sort_mark = "CncFabSortMark";    to my desired parameter. 

Also, I have been able to customize the code for exporting floor parts, wall parts, and structural steel framing elements.  So far it has worked perfectly with every OST family I have tested.   It is a great app!
